src/classes/BGE_BatchGiftEntry_UTIL.cls

```diff
@@ -36,6 +36,82 @@
 */
 public with sharing class BGE_BatchGiftEntry_UTIL {
 
+    /*******************************************************************************************************
+    * @description returns a list of DataImport__c fields the Batch Gift Entry UI needs in SOQL
+    * @param batchId the batch for which to get all fields for soql
+    * @return List<String> list of DataImport__c field api names
+    */
+    public static List<String> getDataImportFields(Id batchId) {
+        return getDataImportFields(batchId, true);
+    }
+
+    /*******************************************************************************************************
+    * @description returns a list of DataImport__c fields the Batch Gift Entry UI needs in SOQL
+    * @param batchId the batch for which to get all fields for soql
+    * @param includeRelationshipFields whether to include relationship fields in the returned list
+    * @return List<String> list of DataImport__c field api names
+    */
+    public static List<String> getDataImportFields(Id batchId, Boolean includeRelationshipFields) {
+
+        List<String> fields = getDataImportFields(includeRelationshipFields);
+        fields.addAll(getActiveFieldNamesFromBatch(batchId));
+
+        return fields;
+    }
+
+    /*******************************************************************************************************
+    * @description returns the subset of DataImport__c fields that are part of every batch
+    * @param includeRelationshipFields whether to include relationship fields in the returned list
```

Comment:
I'm a little confused about why relationship fields would or would not be included. Can you elaborate on this?

Reply:
When BDI_DataImportService uses this list, it wants only the fields directly on DataImport__c. You end up with an error in BDI processing if it tries to parse a `__r.Name` because it tries to get a DFR with that from the UTIL_Describe, and it's an invalid field. 
When BGE uses this list, it needs the related fields in the SOQL query so we can use them in the UI. 
I'm certainly not sure this is the right/best or even a good way to design this. This is a big part of what I wanted to talk through. Looking forward to further thoughts!